src/engine/QueryPlanner.cpp

```diff
@@ -2009,18 +2022,16 @@ std::vector<SubtreePlan> QueryPlanner::createJoinCandidates(
                         ain._qet->getCacheKey() < bin._qet->getCacheKey();
   const auto& a = !swapForTesting ? ain : bin;
   const auto& b = !swapForTesting ? bin : ain;
-  std::vector<SubtreePlan> candidates;
 
-  // TODO<joka921> find out, what is ACTUALLY the use case for the triple
-  // graph. Is it only meant for (questionable) performance reasons
-  // or does it change the meaning.
   JoinColumns jcs;
+  if ((a._idsOfIncludedNodes & b._idsOfIncludedNodes) == 0) {
+    jcs = getJoinColumns(a, b);
+  }
```

Comment:
That looks odd, as it never uses the triple graph. 
Is the condition not rather
`if ( aIncluded & bIncluded !=0) { return {};} // empty result, because can't be joined`

and then the if -else below is exactly as it was?

Reply:
Also I though we agreed on directly returning  `optional<JoinColumns> ` from the `connected` function, 
which also gets an optional tripleGraph, s.t. this function here almost disappears.